lib/bas/bot/humanize_pto.rb

```diff
@@ -75,23 +74,34 @@ def process(read_response)
 
     # write function to execute the PostgresDB write component
     #
-    def write(process_response)
+    def write
       write = Write::Postgres.new(write_options, process_response)
 
       write.execute
     end
 
     private
 
-    def build_params(read_response)
+    def conditions
+      {
+        where: "archived=$1 AND tag=$2 AND stage=$3 ORDER BY inserted_at ASC",
+        params: [false, read_options[:tag], "unprocessed"]
+      }
+    end
+
+    def unprocessable_response
```

Comment:
Could this also be a common utility for certain type of bots? receiving the key for accessing on `read_response.data`?

Reply:
This also depends on the use case, but yes, we can generalize it.